SConstruct

```diff
@@ -910,14 +938,16 @@ if env['system_googletest'] in ('n', 'default'):
 
 # Check for Eigen and checkout submodule if needed
 if env['system_eigen'] in ('y', 'default'):
-    if conf.CheckCXXHeader('Eigen/Dense', '<>'):
+    if conf.CheckCXXHeader('eigen3/Eigen/Dense', '<>'):
```

Comment:
I'm not sure what the right way of handling Eigen is, but this change is insufficient. The Eigen docs suggest that the includes be written as `#include <Eigen/Dense>`, which is what we do, e.g. in `numerics/eigen_dense.h`. If the headers were installed in `/usr/include/eigen3/Eigen` (Ubuntu does this, e.g.), then you could always write `#include <eigen3/Eigen/Dense>`, and pick this up from the fact that `/usr/include` is on the default search path. However, it would not work for other locations for the Eigen headers. The current approach is that to use the system Eigen on such a system, you need to add the right directory to `extra_inc_dirs`, as described in the docstring for the `system_eigen` option.

Reply:
OK that makes sense. The directory is the same on my Mac as well (via Homebrew), so I'm not sure when they would end up in a different directory, but if that's what the docs say, that's what we'll do.

Reply:
I dropped this commit.